libtsuba/include/tsuba/RDG.h

```diff
@@ -162,6 +180,8 @@ class KATANA_EXPORT RDG {
 
   /// name of the graph that was used to load this RDG
   katana::Uri rdg_dir_;
+  /// which partition of the graph was loaded
+  uint32_t partition_number_;
```

Comment:
`uint32_t partition_number_{0}`?

Reply:
This is a little tricky because the real default is `Comm()->ID`. I know we try to have default values for POD types, but is there a way to indicate that the default value should not be used? Could I do `uint32_t partition_number_{-1}`?

Reply:
either `std::optional<uint32_t>` or what you wrote, though a lot of the code base uses `std::numeric_limits<uint32_t>::max()` for `-1` for a little more clarity.